api/src/main/java/com/graphhopper/util/shapes/Polygon.java

```diff
@@ -137,115 +92,55 @@ public boolean contains(Shape s) {
 
     @Override
     public BBox getBounds() {
-        return new BBox(minLon, maxLon, minLat, maxLat);
+        return new BBox(envelope.getMinX(), envelope.getMaxX(), envelope.getMinY(), envelope.getMaxY());
     }
 
     @Override
     public GHPoint getCenter() {
-        return new GHPoint((maxLat + minLat) / 2, (maxLon + minLon) / 2);
+        Point centroid = prepPolygon.getGeometry().getCentroid();
+        return new GHPoint(centroid.getY(), centroid.getX());
     }
 
     @Override
     public double calculateArea() {
-        throw new UnsupportedOperationException("Not supported yet.");
-    }
-
-    private boolean edgesAreIntersecting(
```

Comment:
I think it's great that we can remove this method! This was always a bit of a headache factor for me!